app/validators/unique_in_collection_validator.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# Validator for attributes that only need to be unique within a collection
+# e.g. page.slug within its section, blog_post.slug within the month of posting
+class UniqueInCollectionValidator < ActiveRecord::Validations::UniquenessValidator
+  def validate_each( record, attribute, value )
+    @record = record
```

Comment:
Using instance variables here may not be thread-safe. I *think* validator objects are reused across model objects. We can investigate further on a call to see. In the meantime, consider removing the `@record`, `@attribute`, and `@existing_error_count` instance variables and passing local variables `record`, `attribute`, and `value` to the methods.